src/read/abbrev.rs

```diff
@@ -54,13 +54,25 @@ impl<R: Reader> DebugAbbrev<R> {
     }
 }
 
-impl<R: Reader> Section<R> for DebugAbbrev<R> {
+impl<T> DebugAbbrev<T> {
+    /// Create a `DebugAbbrev` section that references the data in `self`.
+    ///
+    /// This is useful when `R` implements `Reader` but `T` does not.
```

Comment:
An example here would be useful because it isn't otherwise immediately clear to me when one would would need to use this.